### PR TITLE
cython friendly setup.py

### PR DESCRIPTION
**Pull request by yajiedesign**
https://github.com/dmlc/mxnet/pull/4723

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -54,6 +65,9 @@ def config_cython():
     except ImportError:
         print("WARNING: Cython is not installed, will compile without cython module")
         return []
+    except DistutilsPlatformError:
+        print("WARNING: If use Cython. Microsoft Visual C++ is required")
```
Comment (piiswrong) [inline python/setup.py]: How would this know if current vC++ version is compatible with cython?

Comment (yajiedesign) [inline python/setup.py]: ok i am update. the error message will prompt what version vc++.

**Comment by tqchen** [comment]
please check the usage of setup tools and distutils. In general, use setup tools for general build and distutils for inplace build

File: python/setup.py [hunk 2]
```diff
@@ -3,6 +3,11 @@
 from __future__ import absolute_import
 import os
 import sys
+try:
+    from distutils.core import DistutilsPlatformError
+except ImportError:
+    from distutils.errors import DistutilsPlatformError
+
 # need to use distutils.core for correct placement of cython dll
 if "--inplace" in sys.argv:
```
Comment (tqchen) [inline python/setup.py]: see these line. maybe need to import the error into here?

Comment (yajiedesign) [inline python/setup.py]: yes,i see.but DistutilsPlatformError always need to import.

Comment (tqchen) [inline python/setup.py]: I see, thanks

**Pull request closed without merging**